tiatoolbox/annotation/storage.py

```diff
@@ -1071,80 +1072,147 @@ def to_geodict(self) -> Dict[str, Any]:
     def _load_cases(
-        fp: Union[IO, str, Path],
+        full_path: Union[IO, str, Path],
         string_fn: Callable[[Union[str, bytes]], Any],
         file_fn: Callable[[IO], Any],
     ) -> Any:
         with contextlib.suppress(OSError):
-            if isinstance(fp, (Path, str)) and Path(fp).exists():
-                with open(fp) as file_handle:
+            if isinstance(full_path, (Path, str)) and Path(full_path).exists():
+                with open(full_path) as file_handle:
                     return file_fn(file_handle)
-        if isinstance(fp, (str, bytes)):
-            return string_fn(fp)
-        if hasattr(fp, "read"):
-            return file_fn(fp)
+        if isinstance(full_path, (str, bytes)):
+            return string_fn(full_path)
+        if hasattr(full_path, "read"):
+            return file_fn(full_path)
         raise IOError("Invalid file handle or path.")
 
     @classmethod
-    def from_geojson(cls, fp: Union[IO, str]) -> "AnnotationStore":
-        geojson = cls._load_cases(
-            fp=fp,
+    def from_geojson(
+        cls,
+        full_path: Union[IO, str],
+        scale_factor: float = 1,
+        relative_to: Tuple[float, float] = None,
+    ) -> "AnnotationStore":
+        """Create a new database with annotations loaded from a geoJSON file.
+        Args:
+            full_path (Union[IO, str, Path]):
+                The file path or handle to load from.
+            scale_factor (float):
+                The scale factor to use when loading the annotations. All coordinates
+                will be multiplied by this factor to allow import of annotations saved
+                at non-baseline resolution.
+            relative_to [float, float]:
+                The x and y coordinates to use as the origin for the annotations.
+
+        Returns:
+            AnnotationStore:
+                A new annotation store with the annotations loaded from the file.
+
+        """
+        store = cls()
+        store.add_from_geojson(full_path, scale_factor, relative_to=relative_to)
+        return store
+
+    def add_from_geojson(
+        self,
+        full_path: Union[IO, str],
+        scale_factor: float = 1,
+        relative_to: Tuple[float, float] = None,
+    ) -> None:
+        """Add annotations from a .geojson file to an existing store. Make
+        a best effort to create valid shapely geometries from provided contours.
+
+        Args:
+            full_path (Union[IO, str, Path]):
+                The file path or handle to load from.
+            scale_factor (float):
+                The scale factor to use when loading the annotations. All coordinates
+                will be multiplied by this factor to allow import of annotations saved
+                at non-baseline resolution.
+            relative_to [float, float]:
+                The x and y coordinates to use as the origin for the annotations.
+
+        """
+
+        def transform_geom(geom):
```

Comment:
```suggestion
        def transform_geomety(geom):
```

'geom' is not used anywhere else. I think just using the actual word is more clear.